### ISSUE TITLE
Calendar Accessibiity Issue with Keyboard Navigation

### ISSUE THREAD

**Issue by vuefan** [primefaces/primevue#1393]
If you have a PrimeVue PRO Support subscription please post your issue at;

https://pro.primefaces.org 

where our team will respond within 1 business hour.

If you do not have a PrimeVue PRO Support subscription, fill-in the report below. Please note that
your issue will be added to the waiting list of community issues and will be reviewed on a first-come first-serve basis, as a result, the support team is unable to guarantee a specific schedule on when it will be reviewed. Thank you for your understanding.

Current Queue Time for Review
Without PRO Support: ~1-4 weeks.
With PRO Support: 1 hour

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primevue/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=110
```

**CodeSandbox Case (Bug Reports)**
Please demonstrate your case at CodeSandbox by using the issue template below. Issues without a test case have much less possibility to be reviewed in detail and assisted.
WCAG Accessibility Issue

**Current behavior**

In PrimeVue2's Calendar, keyboard navigation works correctly for WCAG accessibility. However, in PrimeVue 3's Calendar, the focus gets lost when trying to navigate with the keyboard as we move the date from one month to the next (in the calendar itself).

**Expected behavior**

Keyboard navigation should function similarly to PrimeVue 2.

**Minimal reproduction of the problem with instructions**

It works correctly in PrimeVue 2 - https://www.primefaces.org/primevue/showcase-v2/#/calendar
It is broken in PrimeVue 3 - https://primefaces.org/primevue/showcase/#/calendar

**What is the motivation / use case for changing the behavior?**

To account for WCAG guidelines for accessibility. At this time, keyboard navigation is broken and users cannot properly select a date using the keyboard.

**Please tell us about your environment:**

Tested on both Windows and Mac in Chrome, Safari, IE.

* **Vue version:** 3.X

Yes 3.x

* **PrimeVue version:** 3.4.X

Yes 3.4.x

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Chrome, Safari, IE
